### PR TITLE
Wrap multiplex with pull-streams

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Work from @dignifiedquire with using multiplex module from npm, wrapping it in pull-streams.

Status: Tests passing!

**Left to do:**

* [ ] Change protocol to `multiplex` instead of `mplex` in both js and go-land
  * [X] JavaScript
  * [ ] Golang - Update: Can't find where this is defined...
* [x] Needs browser test such as - https://github.com/libp2p/js-libp2p-spdy/blob/master/test/browser.js
* [X] Needs to stop running tests with tcp in the browser.
* [ ] Rebase

### CODE REVIEW

File: src/muxer.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  end (cb) {
+    this.multiplex.once('close', cb)
+    this.multiplex.destroy()
+  }
+}
+
+function noop () {}
+
+// Catch error makes sure that even though we get the "Channel destroyed" error from when closing streams, that it's not leaking through since it's not really an error for us, channels shoul close cleanly.
```
Comment (diasdavid) [inline src/muxer.js]: 👌🏾

**Comment by VictorBjelkholm** [comment]
So this is ready to be reviewed now, everything is passing and working. Gonna move on and do the interop testing now. 

**Note:** For this to pass the tests, you'll need to `npm link` both `interface-stream-muxer` from the `fix/pass-error` branch and also use Davids fork of multiplex, `git@github.com:diasdavid/multiplex.git` on the branch `fix/close-muxed-streams`

File: src/multiplex-codec.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+'use strict'
+
+module.exports = '/mplex/6.7.0'
```
Comment (diasdavid) [inline src/multiplex-codec.js]: Should be `multiplex` https://github.com/libp2p/specs/blob/master/7-properties.md#757-protocol-multicodecs, please submit that patch to go-libp2p as well

Comment (VictorBjelkholm) [inline src/multiplex-codec.js]: :+1:

**Comment by diasdavid** [comment]
>  Golang - Update: Can't find where this is defined...

https://github.com/ipfs/go-ipfs/blob/master/core/core.go#L241-L243

**Comment by VictorBjelkholm** [comment]
Closing this in favor of https://github.com/libp2p/js-libp2p-multiplex/pull/53

**Pull request closed without merging**